test/plugins-service.ts

```diff
@@ -419,4 +445,142 @@ describe("Plugins service", () => {
 	});
+	
+	describe("merge xmls tests", () => {
+		let testInjector: IInjector;
+		beforeEach(() => {
+			testInjector = createTestInjector();
+			testInjector.registerCommand("plugin|add", AddPluginCommandLib.AddPluginCommand);
+		});
+		it("fails if the plugin contains incorrect xml", () => {
+			let pluginName = "mySamplePlugin";
+			let projectFolder = createProjectFile(testInjector);
+			let pluginFolderPath = path.join(projectFolder, pluginName);
+			let pluginJsonData = {
+				"name": pluginName,
+				"version": "0.0.1",
+				"nativescript": {
+					"platforms": {
+						"android": "0.10.0"
+					}
+				}
+			};
+			let fs = testInjector.resolve("fs");
+			fs.writeJson(path.join(pluginFolderPath, "package.json"), pluginJsonData).wait();
+			
+			// Adds AndroidManifest.xml file in platforms/android folder
+			createAndroidManifestFile(projectFolder, fs);		
+			
+			// Mock plugins service
+			let pluginsService = testInjector.resolve("pluginsService");
+			pluginsService.getAllInstalledPlugins = () => { 
+				return (() => {
+					return [{
+						name: ""
+					}];
+				}).future<IPluginData[]>()();
+			}
+			
+			let appDestinationDirectoryPath = path.join(projectFolder, "platforms", "android");
+			
+			// Mock platformsData
+			let platformsData = testInjector.resolve("platformsData");
+			platformsData.getPlatformData = (platform: string) => {
+				return {
+					appDestinationDirectoryPath: appDestinationDirectoryPath,
+					frameworkPackageName: "tns-android",
+					configurationFileName: "AndroidManifest.xml"
+				}
+			}
+			
+			// Ensure the pluginDestinationPath folder exists
+			let pluginPlatformsDirPath = path.join(appDestinationDirectoryPath, "app", "tns_modules", pluginName, "platforms", "android");
+			fs.ensureDirectoryExists(pluginPlatformsDirPath).wait();
+			
+			// Creates invalid plugin's AndroidManifest.xml file
+			let xml = '<?xml version="1.0" encoding="UTF-8"?>' +
+				'<manifest xmlns:android="http://schemas.android.com/apk/res/android" package="com.example.android.basiccontactables" android:versionCode="1" android:versionName="1.0" >' + 
+    			'<uses-permission android:name="android.permission.READ_CONTACTS"/>';
+			let pluginConfigurationFilePath = path.join(pluginPlatformsDirPath, "AndroidManifest.xml");
+			fs.writeFile(pluginConfigurationFilePath, xml).wait();
+				
+			// Expected error message. The assertion happens in mockBeginCommand
+			let expectedErrorMessage = `Exception: Invalid xml file ${pluginConfigurationFilePath}. Additional technical information: element parse error: Exception: Invalid xml file ` + 
+				`${pluginConfigurationFilePath}. Additional technical information: unclosed xml attribute` + 
+				`\n@#[line:1,col:39].` + 
+				`\n@#[line:1,col:39].`;
+			mockBeginCommand(testInjector, expectedErrorMessage);				
+			
+			let commandsService = testInjector.resolve(CommandsServiceLib.CommandsService);		
+			commandsService.tryExecuteCommand("plugin|add", [pluginFolderPath]).wait();
```

Comment:
Please reconsider leaving the test as-is. Currently you are testing the entire command infrastructure and the plugins service at the same time, which is not the correct way of writing unit tests.